### PR TITLE
Add question definitions to GitHub

### PR DESCRIPTION
**Pull request by ronin1972**

### CODE REVIEW

File: agreements/EventDefinition/EventJSON/eoI.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+                        "dataType":"date-time",
```
Comment (tommyb82) [inline agreements/EventDefinition/EventJSON/eoI.json:1]: Looks like we're missing part of the file here?

**Review by tommyb82 — CHANGES_REQUESTED** [review]
EOI template seems to be part missing

File: agreements/EventDefinition/EventJSON/eoI.json (hunk 1 above)
Comment (adrianmilne) [reply agreements/EventDefinition/EventJSON/eoI.json:1]: I think this file is already out of date now too - we probably need to overwrite with @sudhakar-reddy-konda 's latest version

Comment (ronin1972) [reply agreements/EventDefinition/EventJSON/eoI.json:1]: I realised a couple of days ago that something was missing. Just not had time to fix. Think I might have time now. Where is the most up to date version? Still in the SQL script but just an updated one?

Comment (ronin1972) [reply agreements/EventDefinition/EventJSON/eoI.json:739]: @sudhakar-reddy-konda Updated with what I think is the latest version

**Comment by adrianmilne** [comment]
@ronin1972  - I was playing around with a slightly different approach to this - of having the data templates in the ccs-scale-db-scripts-data repo. Managed to get that working so we could reference the external scripts directly from the `.sql` files - so removing the need to maintain duplicate copies across repos. That's now committed and working.

https://github.com/Crown-Commercial-Service/ccs-scale-db-scripts-data/tree/develop/agreements/data

If you're happy with that approach, this PR could be closed (or the data templates removed, and just the event.yaml maintained in here)

File: cat/CaT-service.yaml [hunk 2]
```diff
@@ -3886,10 +3861,105 @@ components:
...
+                  description: |+
+                    Describes the relationships the answers provided to a question.
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      dependentOnId:
+                        type: string
+                      relationshipType:
+                        $ref: '#/components/schemas/DependencyType'
```
Comment (adrianmilne) [inline cat/CaT-service.yaml:3893]: Should this be `#/components/schemas/RelationshipType`?

**Review by sudhakar-reddy-konda — APPROVED** [review]

File: cat/CaT-service.yaml [hunk 3]
```diff
@@ -3929,15 +3910,124 @@ components:
...
+    QuestionDependancy:
+      type: object
+      description: |+
+        Describes the condition when this question should be shown based on another question.
+      properties:
+        dependentOnId:
+          type: string
+        dependencyType:
+          $ref: '#/components/schemas/DependencyType'
+        dependencyValue:
```
Comment (tommyb82) [inline cat/CaT-service.yaml:3974]: This block is causing errors in the swagger editor:
```
Structural error at components.schemas.QuestionDependancy.properties.dependencyValue.type
should be string
Jump to line 3975
Structural error at components.schemas.QuestionDependancy.properties.dependencyValue.type
should be equal to one of the allowed values
allowedValues: array, boolean, integer, number, object, string
Jump to line 3975
```

**Review by tommyb82 — CHANGES_REQUESTED** [review]
Swagger editor errors

**Review by tommyb82 — APPROVED** [review]
Approved - but we can't merge this until corresponding Tenders API changes are also lined up as this introduces compilation/build errors.

FYI @ronin1972 @sudhakar-reddy-konda @adrianmilne

**Pull request merged**